src/agent/agent.c

```diff
@@ -2802,6 +2780,71 @@ static bool agent_connect(Agent *agent) {
                 }
         }
 
+        return 0;
+}
+
+static int agent_register_callback(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
+        Agent *agent = (Agent *) userdata;
+
+        if (agent->connection_state != AGENT_CONNECTION_STATE_CONNECTING) {
+                bc_log_error("Agent is not in CONNECTING state, dropping Register callback");
+                if (agent->register_call_slot != NULL) {
+                        sd_bus_slot_unrefp(&agent->register_call_slot);
+                        agent->register_call_slot = NULL;
+                }
+                return -EPERM;
+        }
```

Comment:
Actually we can drop the `AGENT_CONNECTION_STATE_CONNECTING` with just checking if `register_call_slot` is not NULL. But I thought that it is more clear to have another state for the `connection_state` since it is really a separate state.

We can drop the NULL check though

Reply:
Dropped the NULL check